File: app/util/java/NoShowHandler.java
package util.java;

import com.google.inject.ImplementedBy;
import models.ExamEnrolment;

import java.util.List;

@ImplementedBy(NoShowHandlerUtil.class)
public interface NoShowHandler {

    void handleNoShows(List<ExamEnrolment> noShows, Class<?> sender);
    void handleNoShow(ExamEnrolment noShow);
    void handleNoShowAndNotify(ExamEnrolment enrolment);

}

File: app/services/enrolment/NoShowHandler.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package services.enrolment

import com.google.inject.ImplementedBy
import models.enrolment.{ExamEnrolment, Reservation}

@ImplementedBy(classOf[NoShowHandlerImpl])
trait NoShowHandler {
  def handleNoShows(noShows: List[ExamEnrolment], reservations: List[Reservation]): Unit
  def handleNoShowAndNotify(enrolment: ExamEnrolment): Unit
}
